File: solutions/dart/high-scores/1/lib/high_scores.dart
class HighScores {
  List<int> _scores = [];
  List<int> _sorted_scores = [];

  HighScores(List<int> scores) {
    this._scores = List.of(scores);
    this._sorted_scores = List.of(scores);
    this._sorted_scores.sort((a, b) => b - a);
  }

  List<int> get scores {
    return List.of(this._scores);
  }

  int latest() {
    return this._scores.last;
  }

  int personalBest() {
    return this._sorted_scores.first;
  }

  List<int> personalTopThree() {
    return List.of(this._sorted_scores.take(3));
  }
}

File: solutions/javascript/high-scores/1/high-scores.js

export class HighScores {
  
  constructor(scores=[]) {
    this._scores = scores;
  }

  get scores() {
    return [...this._scores];
  }

  get latest() {
    if (this._scores.length > 0) return this._scores[this._scores.length-1];
  }

  get personalBest() {
    return this._scores.reduce((acc,val) => val > acc? val : acc, 0);
  }

  get personalTopThree() {
    // this.scores returns a copy, so the sort doesn't perturbe the recorded scores order.
    let descendingScores = this.scores.sort((a,b) => b - a);
    let lim = Math.min(3, descendingScores.length);
    return descendingScores.slice(0, 3);
  }
}
